framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -997,6 +1024,44 @@ private static class ClassOrInterfaceAnnos {
     /** Collection of declared enum constants (empty if not an enum). */
     public Set<String> enumConstants = new HashSet<>(2);
 
+    /**
+     * Annotations on the declaration of the class (note that despite the name, these can also be
+     * type annotations).
+     */
+    private @MonotonicNonNull Set<AnnotationMirror> declarationAnnotations = null;
+
+    /**
+     * The Java Parser TypeDeclaration representing the class's declaration. Used for placing
+     * annotations inferred on the class declaration itself.
+     */
+    private @MonotonicNonNull TypeDeclaration<?> declaration = null;
+
+    /**
+     * Create a new ClassOrInterfaceAnnos
+     *
+     * @param javaParserNode the java parser node corresponding to the class declaration, which is
+     *     used for placing annotations on the class declaration
+     */
+    public ClassOrInterfaceAnnos(@Nullable TypeDeclaration<?> javaParserNode) {
+      if (javaParserNode != null) {
```

Comment:
What is the point of this test?  I would drop it, and the `= null` where `declaration` is declared.